--- libtorrent-rasterbar-0.15.0/configure.ac~	2010-02-08 06:18:10.000000000 +0100
+++ libtorrent-rasterbar-0.15.0/configure.ac	2010-03-22 09:59:00.075165939 +0100
@@ -390,7 +390,7 @@
   ["yes"|"default"], [
       AC_MSG_RESULT([yes])
       AC_DEFINE([TORRENT_LOGGING],[1],[Define to enable logging of the session events.])
-      COMPILETIME_OPTIONS+="-DTORRENT_LOGGING "
+      COMPILETIME_OPTIONS="${COMPILETIME_OPTIONS} -DTORRENT_LOGGING "
     ],
   ["no"|"none"], [
       AC_MSG_RESULT([no])
@@ -398,12 +398,12 @@
   ["verbose"], [
       AC_MSG_RESULT([verbose])
       AC_DEFINE([TORRENT_VERBOSE_LOGGING],[1],[Define to enable logging of the session events and every peer connection.])
-      COMPILETIME_OPTIONS+="-DTORRENT_VERBOSE_LOGGING "
+      COMPILETIME_OPTIONS="${COMPILETIME_OPTIONS} -DTORRENT_VERBOSE_LOGGING "
     ],
   ["errors"], [
       AC_MSG_RESULT([errors])
       AC_DEFINE([TORRENT_ERROR_LOGGING],[1],[Define to enable logging of the session events and every peer connection limited to errors.])
-      COMPILETIME_OPTIONS+="-DTORRENT_ERROR_LOGGING "
+      COMPILETIME_OPTIONS+="${COMPILETIME_OPTIONS} -DTORRENT_ERROR_LOGGING "
     ],
   [AC_MSG_RESULT([$ARG_ENABLE_LOGGING])
    AC_MSG_ERROR([Unknown option "$ARG_ENABLE_LOGGING". Use either "yes", "no", "verbose" or "errors".])]
@@ -446,7 +446,7 @@
 
       AX_CHECK_OPENSSL([
           AC_DEFINE([TORRENT_USE_OPENSSL],[1],[Define to use OpenSSL support.])
-          COMPILETIME_OPTIONS+="-DTORRENT_USE_OPENSSL "
+          COMPILETIME_OPTIONS="${COMPILETIME_OPTIONS} -DTORRENT_USE_OPENSSL "
         ], [
           AC_MSG_ERROR([OpenSSL library not found. Try using --with-openssl=DIR or disabling encryption at all.])
       ])
@@ -454,7 +454,7 @@
   ["no"|"off"], [
       AC_MSG_RESULT([no])
       AC_DEFINE([TORRENT_DISABLE_ENCRYPTION],[1],[Define to disable any encryption support and avoid linking against OpenSSL.])
-      COMPILETIME_OPTIONS+="-DTORRENT_DISABLE_ENCRYPTION "
+      COMPILETIME_OPTIONS="${COMPILETIME_OPTIONS} -DTORRENT_DISABLE_ENCRYPTION "
     ],
   [AC_MSG_RESULT([$ARG_ENABLE_ENCRYPTION])
    AC_MSG_ERROR([Unknown option "$ARG_ENABLE_ENCRYPTION". Use either "yes" or "no".])]
@@ -468,7 +468,7 @@
   ["no"], [
     AC_MSG_RESULT([no])
     AC_DEFINE([TORRENT_DISABLE_GEO_IP],[1],[Define to disable the GeoIP support and avoid linking against LGPLed code.])
-    COMPILETIME_OPTIONS+="-DTORRENT_DISABLE_GEO_IP "
+    COMPILETIME_OPTIONS+="${COMPILETIME_OPTIONS} -DTORRENT_DISABLE_GEO_IP "
    ],
   [AC_MSG_RESULT([$ARG_ENABLE_GEOIP])
    AC_MSG_ERROR([Unknown option "$ARG_ENABLE_GEOIP". Use either "yes" or "no".])]
@@ -482,12 +482,12 @@
   ["no"|"off"], [
       AC_MSG_RESULT([no])
       AC_DEFINE([TORRENT_DISABLE_DHT],[1],[Define to disable the DHT support.])
-      COMPILETIME_OPTIONS+="-DTORRENT_DISABLE_DHT "
+      COMPILETIME_OPTIONS="${COMPILETIME_OPTIONS} -DTORRENT_DISABLE_DHT "
     ],
   ["logging"], [
       AC_MSG_RESULT([logging])
       AC_DEFINE([TORRENT_DHT_VERBOSE_LOGGING],[1],[Define to enable DHT support with verbose logging.])
-      COMPILETIME_OPTIONS+="-DTORRENT_DHT_VERBOSE_LOGGING "
+      COMPILETIME_OPTIONS="${COMPILETIME_OPTIONS} -DTORRENT_DHT_VERBOSE_LOGGING "
     ],
   [AC_MSG_RESULT([$ARG_ENABLE_DHT])
    AC_MSG_ERROR([Unknown option "$ARG_ENABLE_DHT". Use either "yes", "no" or "logging".])]
@@ -627,10 +627,10 @@
 # this works around a bug in asio in boost-1.39
 # see: https://svn.boost.org/trac/boost/ticket/3095
 AC_DEFINE([BOOST_ASIO_HASH_MAP_BUCKETS],[1021],[Define to fix a wrong behavior in boost 1.39.])
-COMPILETIME_OPTIONS+="-DBOOST_ASIO_HASH_MAP_BUCKETS=1021 "
+COMPILETIME_OPTIONS="${COMPILETIME_OPTIONS} -DBOOST_ASIO_HASH_MAP_BUCKETS=1021 "
 
 AC_DEFINE([BOOST_EXCEPTION_DISABLE],[1],[Define to disable the boost.exception features.])
-COMPILETIME_OPTIONS+="-DBOOST_EXCEPTION_DISABLE "
+COMPILETIME_OPTIONS+="${COMPILETIME_OPTIONS} -DBOOST_EXCEPTION_DISABLE "
 
 dnl Use possibly specific python install params
 AC_ARG_VAR([PYTHON_INSTALL_PARAMS], [Set specific install parameters for python bindings.])
@@ -640,7 +640,7 @@
 dnl Set some defines if we are building a shared library
 AS_IF([test "x$enable_shared" = "xyes"],
       [AC_DEFINE([TORRENT_BUILDING_SHARED],[1],[Define to exports functions and classes with default visibility in GCC.])
-       COMPILETIME_OPTIONS+="-DTORRENT_LINKING_SHARED "])
+       COMPILETIME_OPTIONS="${COMPILETIME_OPTIONS} -DTORRENT_LINKING_SHARED "])
 
 AC_SUBST(DEBUGFLAGS)
 AC_SUBST(PYTHON_INSTALL_PARAMS)
