--- libtorrent-rasterbar-0.13.1-orig/src/torrent_info.cpp	2008-07-01 18:03:01.000000000 +0200
+++ libtorrent-rasterbar-0.13.1/src/torrent_info.cpp	2008-08-30 19:00:05.000000000 +0200
@@ -354,7 +354,11 @@
   		{
  			m_name = tmp.leaf();
   		}
- 		else if (tmp.has_branch_path())
+#if BOOST_VERSION < 103600
+		else if (tmp.has_branch_path())
+#else
+		else if (tmp.has_parent_path())
+#endif
   		{
  			fs::path p;
  			for (fs::path::iterator i = tmp.begin()
@@ -572,7 +576,11 @@
 	{
 //		TORRENT_ASSERT(file.begin() != file.end());
 
-		if (!file.has_branch_path())
+#if BOOST_VERSION < 103600
+		if (!file.has_branch_path())
+#else
+		if (!file.has_parent_path())
+#endif
 		{
 			// you have already added at least one file with a
 			// path to the file (branch_path), which means that
@@ -662,7 +670,11 @@
 					fs::path const* file_path;
 					if (i->orig_path) file_path = &(*i->orig_path);
 					else file_path = &i->path;
+#if BOOST_VERSION < 103600
 					TORRENT_ASSERT(file_path->has_branch_path());
+#else
+					TORRENT_ASSERT(file_path->has_parent_path());
+#endif
 					TORRENT_ASSERT(*file_path->begin() == m_name);
 
 					for (fs::path::iterator j = boost::next(file_path->begin());
