--- libtorrent-rasterbar-0.13.1-orig/configure.in	2008-07-03 13:32:18.000000000 +0200
+++ libtorrent-rasterbar-0.13.1/configure.in	2008-08-30 15:29:49.000000000 +0200
@@ -393,10 +393,10 @@
 AC_SUBST(COMPILETIME_OPTIONS)
 
 AC_CONFIG_FILES(Makefile src/Makefile include/Makefile libtorrent-rasterbar.pc)
-if [[ "x$examples" == "xyes" ]]; then
-	AC_CONFIG_FILES(examples/Makefile)
-fi
-if [[ "x$tests" == "xyes"  ]]; then
-	AC_CONFIG_FILES(test/Makefile)
-fi
+# if [[ "x$examples" == "xyes" ]]; then
+#	AC_CONFIG_FILES(examples/Makefile)
+# fi
+# if [[ "x$tests" == "xyes"  ]]; then
+#	AC_CONFIG_FILES(test/Makefile)
+# fi
 AC_OUTPUT
