--- libtorrent-rasterbar-0.14.10/m4/ax_python-fixed.m4.wiget	2010-07-09 23:38:46.485060875 +0200
+++ libtorrent-rasterbar-0.14.10/m4/ax_python-fixed.m4	2010-07-09 23:39:15.082559709 +0200
@@ -58,7 +58,7 @@
 AC_DEFUN([AX_PYTHON],
 [AC_MSG_CHECKING(for python build information)
 AC_MSG_RESULT([])
-for python in python2.6 python2.5 python2.4 python2.3 python2.2 python2.1 python; do
+for python in python2.7 python2.6 python2.5 python2.4 python2.3 python2.2 python2.1 python; do
 # AC_CHECK_PROGS(PYTHON_BIN, [$python])
 PYTHON_BIN=$python
 ax_python_bin=$PYTHON_BIN
